metamorph/processor_test.go

```diff
@@ -621,15 +621,14 @@ func BenchmarkProcessTransaction(b *testing.B) {
 
 		txs[txID] = &txHash
 
-		processor.ProcessTransaction(NewProcessorRequest(
-			context.Background(),
-			&store.StoreData{
+		processor.ProcessTransaction(&ProcessorRequest{
```

Comment:
Why not use the constructor as before `NewProcessorRequest`

Reply:
the constructor does not actually do anything except initialising structure and the arguments seem less readable. 
Also you don't have to specify default value for channels for example.

Reply:
also from 
https://pkg.go.dev/context:
"Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:"